### PR TITLE
Add more tests for [Epic] Better communicate embedding options in Admin settings (#23498)

### PR DESCRIPTION
**Pull request by WiNloSt**
Related to #23498

### CODE REVIEW

File: frontend/test/metabase/scenarios/sharing/public.cy.spec.js [hunk 1]
```diff
@@ -21,10 +22,7 @@ const USERS = {
   "anonymous user": () => cy.signOut(),
 };
 
-// [quarantine]: failing almost consistently in CI
-// Skipping the whole spec because it needs to be refactored.
-// If possible, re-use as much code as possible but let test run in isolation.
-describe.skip("scenarios > public", () => {
+describe("scenarios > public", () => {
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/sharing/public.cy.spec.js:25]: @nemanjaglumac this was skipped in https://github.com/metabase/metabase/pull/13945 which is almost 2 years ago, can you help review if this is good now or should it still be disabled?

I tested locally > 10 times and it's still going strong in green.

File: frontend/src/metabase/admin/settings/selectors.js [hunk 2]
```diff
@@ -313,6 +313,7 @@ const SECTIONS = updateSectionsWithPlugins({
       {
         key: "enable-public-sharing",
         display_name: t`Enable Public Sharing`,
+        description: t`Enable admins to create publicly viewable links (and embeddable iframes) for Questions and Dashboards.`,
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/settings/selectors.js:316]: Update copy [according to the design](https://www.figma.com/file/SAi6pRHuhrWeVRuOHH4evg/Better-communicate-embedding-options-in-Admin-settings?node-id=27%3A996).

File: frontend/test/metabase/scenarios/models/helpers/e2e-models-helpers.js [hunk 3]
```diff
@@ -61,7 +61,7 @@ export function selectDimensionOptionFromSidebar(name) {
 }
 
 export function openDetailsSidebar() {
-  cy.findByTestId("saved-question-header-button").click();
+  cy.findByTestId("saved-question-header-title").click();
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/models/helpers/e2e-models-helpers.js:64]: The test ID is changed in this commit. https://github.com/metabase/metabase/commit/ee25189dd8073dc7e4a51b0a0ab15fcade3bf7ce#diff-f66769a8ce4d54035a928c1eb05e282e1277a7e6c582139b0218f73160066506R36

File: frontend/test/metabase/scenarios/organization/edit-history.-metadatacy.spec.js [hunk 4]
```diff
@@ -88,7 +88,7 @@ function changeDashboard() {
 
 function changeQuestion() {
   cy.intercept("PUT", "/api/card/**").as("updateQuestion");
-  cy.findByTestId("saved-question-header-button").click();
+  cy.findByTestId("saved-question-header-title").click();
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/organization/edit-history.-metadatacy.spec.js:91]: The test ID is changed in this commit. https://github.com/metabase/metabase/commit/ee25189dd8073dc7e4a51b0a0ab15fcade3bf7ce#diff-f66769a8ce4d54035a928c1eb05e282e1277a7e6c582139b0218f73160066506R36

File: frontend/test/metabase/scenarios/sharing/public.cy.spec.js [hunk 5]
```diff
@@ -159,6 +160,22 @@ describe.skip("scenarios > public", () => {
         });
     });
 
+    it("should show shared questions and dashboards in admin settings", () => {
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/sharing/public.cy.spec.js:163]: Test the public sharing admin page.

File: frontend/test/metabase/scenarios/sharing/public.cy.spec.js [hunk 6]
```diff
@@ -67,14 +64,18 @@ describe.skip("scenarios > public", () => {
     it("should allow users to create parameterized dashboards", () => {
       visitQuestion(questionId);
 
-      cy.findByTestId("saved-question-header-button").click();
-      cy.findByTestId("add-to-dashboard-button").click();
+      openQuestionActions();
+
+      popover().within(() => {
+        cy.findByText("Add to dashboard").click();
+      });
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/sharing/public.cy.spec.js:71]: We moved **Add to dashboard** to question actions already.

File: frontend/test/metabase/scenarios/sharing/public.cy.spec.js [hunk 7]
```diff
@@ -67,14 +64,18 @@ describe.skip("scenarios > public", () => {
...
+
       modal()
         .contains("Create a new dashboard")
         .click();
       modal()
         .get('input[name="name"]')
-        .type("parameterized dashboard");
+        .type("parameterized dashboard", { delay: 0 });
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/sharing/public.cy.spec.js:78]: Speed up the test a bit.

File: frontend/test/metabase/scenarios/sharing/public.cy.spec.js [hunk 8]
```diff
@@ -111,7 +112,7 @@ describe.skip("scenarios > public", () => {
       cy.contains(COUNT_DOOHICKEY);
 
       cy.url()
-        .should("match", /\/dashboard\/\d+\?text=Doohickey$/)
+        .should("match", /\/dashboard\/\d+[-\w]+\?text=Doohickey$/)
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/sharing/public.cy.spec.js:115]: I suspect we've changed the URL in the past 2 years, so I updated the regex.

File: frontend/src/metabase/admin/settings/components/widgets/EmbeddingCustomizationInfo.tsx [hunk 9]
```diff
@@ -6,9 +6,9 @@ import SettingHeader from "../SettingHeader";
 
 export const EmbeddingCustomizationInfo = () => {
   const setting = {
-    description: jt`In order to remove the Metabase logo from embeds, you can always upgrade to one of our ${(
+    description: jt`In order to remove the Metabase logo from embeds, you can always upgrade to ${(
       <ExternalLink href={MetabaseSettings.upgradeUrl()}>
-        {t`paid plans.`}
+        {t`one of our paid plans.`}
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/settings/components/widgets/EmbeddingCustomizationInfo.tsx:11]: [See the Slack discussion for context.](https://metaboat.slack.com/archives/C03E8MQJZBM/p1657108198925169?thread_ts=1657093586.836219&cid=C03E8MQJZBM)

File: frontend/src/metabase/admin/settings/components/widgets/EmbeddingCustomizationInfo.tsx [hunk 10]
```diff
@@ -6,9 +6,9 @@ import SettingHeader from "../SettingHeader";
 
 export const EmbeddingCustomizationInfo = () => {
   const setting = {
-    description: jt`In order to remove the Metabase logo from embeds, you can always upgrade to one of our ${(
+    description: jt`In order to remove the Metabase logo from embeds, you can always upgrade to ${(
       <ExternalLink href={MetabaseSettings.upgradeUrl()}>
```
Comment (ranquild) [inline frontend/src/metabase/admin/settings/components/widgets/EmbeddingCustomizationInfo.tsx:10]: @WiNloSt the way `jt` works is that is requires using `key` for all jsx items, e.g. `<ExternalLink key="link"`. The key should be unique among other items in the enclosing `jt` expression. It also works without the `key` but it prints warnings to the console.

**Review by ranquild — APPROVED** [review]
Looks good. Please take a look at the comments.

File: frontend/src/metabase/admin/settings/components/widgets/EmbeddingCustomizationInfo.tsx (hunk 10 above)
Comment (WiNloSt) [reply frontend/src/metabase/admin/settings/components/widgets/EmbeddingCustomizationInfo.tsx:10]: Dang! I never knew. I'll fix those.

File: frontend/test/metabase/scenarios/sharing/public.cy.spec.js [hunk 11]
```diff
@@ -145,6 +154,22 @@ describe.skip("scenarios > public", () => {
...
+      cy.visit("/admin/settings/public_sharing");
+
+      cy.findByText("Enable Public Sharing");
+
+      cy.findByText(
+        "Enable admins to create publicly viewable links (and embeddable iframes) for Questions and Dashboards.",
+      );
+
+      // shared questions
+      cy.findByText("sql param");
```
[the remark this answers is not in the document]
Comment (WiNloSt) [reply frontend/test/metabase/scenarios/sharing/public.cy.spec.js]: Fixed

File: frontend/src/metabase/admin/settings/components/widgets/EmbeddingCustomizationInfo.tsx (hunk 10 above)
Comment (WiNloSt) [reply frontend/src/metabase/admin/settings/components/widgets/EmbeddingCustomizationInfo.tsx]: Fixed.

**Pull request merged**